decentraland/unity-renderer
# Chore: double call in explore v2 on select

## What does this PR change?
Closes #3030 

### Problem Description
Opening or switching **ExploreV2Menu** sections from UI toggle or via keyboard shortcut was triggering `OnSectionOpen` event of `ExploreV2MenuComponentView` several times in one frame. Real problem is the chain calls between View and Controller for opening section task. That should be treated separately as a task of bigger refactoring of this pair (with more clear separation of responsibilities).

### Solution
- To eliminate event to be called several time per frame I added a simple lock variable `isOpeningSectionThisFrame `(line 39 in `ExploreV2MenuComponentView`) and a coroutine `ResetSectionOpenLock()` (line 164) which updates it in one frame after a call (they used inside a logic of `OnSectionSelected` function).
- Add checks of variables change to apply it only if values changes - for visibility vars inside View's `ChangeVisibilityVarForSwitchedSections` (line 255), and for `SectionToggle.SelectToggle` (line 156)


### Additional changes
- Applied code style to both Controller and View and rearranged the code
- extracted Realm logic to a separate `ExploreV2ComponentRealmsController` class
- DRY refactored related Tests via applying [TestCases]

## How to test the changes?
Not meant for qa. Devs only

1. Checkout the branch
2. Go to the one commit before the [removed debugs](https://github.com/decentraland/unity-renderer/pull/3896/commits/cde3e80e5983206ae90713f8444fd7f77cc82d48)
3. Run from Editor. 
4. Open Explorer - via UI and via keyboard
5. Observe only 1 SHOW logs printed in the console
6. Switch between tabs - and observe as well 1 SHOW logs printed for the event

if want to see 3 SHOW logs - switch to some earlier commits on the branch

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
